sonar-core/src/main/resources/org/sonar/core/component/db/ComponentMapper.xml

```diff
@@ -227,5 +227,30 @@
     delete from projects where kee=#{key}
   </delete>
 
-</mapper>
+  <select id="selectProvisionedProjects" parameterType="map" resultType="Component">
+    select <include refid="componentColumns"/>
+    from projects p
+    <include refid="provisionClauses"/>
+  </select>
 
+  <select id="countProvisionedProjects" parameterType="map" resultType="int">
+    select count(p.id)
+    from projects p
+    <include refid="provisionClauses"/>
+  </select>
+
+  <sql id="provisionClauses">
+    left join snapshots s on s.project_id=p.id
+    where
+    s.id is null
```

Comment:
not so sure about it. There's pro's and con's. In the join statements you're supposed to put join conditions. In the where clause, the filtering occurs. That's how I separate responsabilities